sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

```diff
@@ -0,0 +1,547 @@
+the kernel is submitted to a non-FPGA device.
+
+All of the properties defined in this extension are meaningful only on the
+kernel argument and are not meaningful within the kernel body.
+
+The properties that are defined for pointers will be meaningful for
+`annotated_arg<T, ...>` only when T is a pointer type.
+
+```c++
+namespace sycl::ext::intel::experimental {
+struct conduit_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    conduit_key>;
+};
+
+struct register_map_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    register_map_key>;
+};
+
+struct stable_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    stable_key>;
+};
+
+struct buffer_location_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    buffer_location_key, std::integral_constant<int, K>>;
+};
+
+struct awidth_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    awidth_key, std::integral_constant<int, K>>;
+};
+
+struct dwidth_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    dwidth_key, std::integral_constant<int, K>>;
+};
+
+enum class read_write_mode_enum {
+  read_write,
+  read,
+  write
+};
+
+struct read_write_mode_key {
+  template<read_write_mode_enum mode>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    read_write_mode_key, std::integral_constant<read_write_mode_enum, mode>>;
+};
+
+struct latency_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    latency_key, std::integral_constant<int, K>>;
+};
+
+struct maxburst_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    maxburst_key, std::integral_constant<int, K>>;
+};
+
+struct wait_request_key {
+  template<bool K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    wait_request_key, std::integral_constant<bool, K>>;
+};
+
+inline constexpr conduit_key::value_t
+ conduit;
+inline constexpr register_map_key::value_t
+ register_map;
+inline constexpr stable_key::value_t stable;
+template<int K> inline constexpr buffer_location_key::value_t<K>
+ buffer_location;
+template<int K> inline constexpr awidth_key::value_t<K>
+ awidth;
+template<int K> inline constexpr dwidth_key::value_t<K>
+ dwidth;
+template<read_write_mode_enum mode>
+inline constexpr read_write_mode_key::value_t<mode>
+ read_write_mode;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::read>  read_write_mode_read;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::write>  read_write_mode_write;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::read_write>
+    read_write_mode_readwrite;
```

Comment:
`read_write_mode_readwrite` is a really bad name.

Would it make sense to split this out into separate `read`, `write` and `read_write` properties?